### PR TITLE
Add support for Split.ALL

### PR DESCRIPTION
**Pull request by mariosasko**
The title says it all.

### CODE REVIEW

File: src/datasets/arrow_reader.py [hunk 1]
```diff
@@ -105,6 +105,8 @@ def make_file_instructions(name, split_infos, instruction, filetype_suffix=None)
     """
     name2len = {info.name: info.num_examples for info in split_infos}
     if not isinstance(instruction, ReadInstruction):
+        if str(instruction) == "all":
+            instruction = "+".join(name2len.keys())
```
Comment (lhoestq) [inline src/datasets/arrow_reader.py:109]: we may need to sort the keys here to make the order deterministic

**Review by lhoestq — COMMENTED** [review]
Awesome thank you !
I left one minor comment

File: src/datasets/arrow_reader.py (hunk 1 above)
Comment (mariosasko) [reply src/datasets/arrow_reader.py:109]: `dict` preserves insertion order in Python 3.6+, so the ordering of the keys is deterministic (this is an implementation detail in CPython in 3.6 and was documented in 3.7).

**Comment by mariosasko** [comment]
Honestly, I think we should fix some other issues in Split API before this change. E. g. currently the following will not work, even though it should:
```python
import datasets
datasets.load_dataset("sst", split=datasets.Split.TRAIN+datasets.Split.TEST)  # AssertionError
```

**Review by lhoestq — APPROVED** [review]
Right, merging this one then :)
thank you !

**Pull request merged**